contracts/azorius/LinearERC721Voting.sol

```diff
@@ -0,0 +1,292 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity =0.8.19;
+
+import { IERC721 } from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import { BaseVotingBasisPercent } from "./BaseVotingBasisPercent.sol";
+import { IAzorius } from "./interfaces/IAzorius.sol";
+import { BaseStrategy } from "./BaseStrategy.sol";
+
+/**
+ * Allows multiple ERC721 tokens to be registered on the strategy as governance tokens, 
+ * each with their own voting weight.
+ *
+ * Since there is no way to snapshot ERC721 holdings, each ERC721 id can vote once, reguardless
+ * of where it was when a proposal was created.
+ *
+ * Also, this uses "quorumThreshold" rather than quorumPercent, because total supply is not knowable
+ * within the IERC721 interface.  This is similar to a multisig "total signers" required, rather than
+ * a percentage of the tokens.
+ */
+contract LinearERC721Voting is BaseStrategy, BaseVotingBasisPercent {
+
+    struct GovernanceNFT {
+        IERC721 token;
+        uint256 weight;
+        bool isProposer;
+    }
+
+    GovernanceNFT[] public governanceTokens;
```

Comment:
I feel like this storage pattern may be more efficient, to prevent having to loop through an array when counting weight:

`
   struct GovernanceNFT {
        uint256 weight;
        bool isProposer;
    }

   mapping(address => GovernanceNFT) public governanceTokens;
`

Reply:
oh yeah awesome, we wouldn't really need an array of governance addresses, because voting always involves giving those addresses

I'll have to think through that through but this is probably way better

Reply:
hmm, @tbwebb22 we still need a way to check if an individual address can create proposals, in `isProposer(address _address)`
    
    I'm not sure we can do that without keeping an array of NFT addresses and seeing if the personal address has a balance